### PR TITLE
chore: use larger runners for long checks

### PR DESCRIPTION
**Pull request by wallyworld**
We now have `quad-xlarge` runners. Using these for the smoke, upgrade, snapcraft and terraform check jobs saves approx 5 minutes or more off the execution time.

The upgrade test on k8s was failing due to a bad upstream charm (prometheus-k8s). Instead, just use snappass-test which is simpler and we control it.

Finally, when bootstrapping lxd, use the enable-os-refresh=false option to save some time.

[JIRA-6335](https://warthogs.atlassian.net/browse/JUJU-6335)

### CODE REVIEW

File: .github/workflows/migrate.yml [hunk 1]
```diff
@@ -82,7 +82,9 @@ jobs:
       - name: Bootstrap target controller
         run: |
           juju version
-          juju bootstrap lxd target-controller --constraints "arch=$(go env GOARCH)"
+          juju bootstrap lxd target-controller \
+            --constraints "arch=$(go env GOARCH)" \
+            --model-default enable-os-upgrade=false
```
Comment (SimonRichardson) [inline .github/workflows/migrate.yml:87]: This will go away in 4.0, but it just shows that it's not required.

**Review by SimonRichardson — APPROVED** [review]

**Comment by SimonRichardson** [comment]
> === CONT  TestAcc_DataSourceMachine_Stable
> cannot run Terraform provider tests: error calling terraform version command: fork/exec /tmp/plugintest-terraform272785777/terraform: text file busy
> FAIL	github.com/juju/terraform-provider-juju/internal/provider	119.609s

That seems like there is a bug in terraform.

**Comment by wallyworld** [comment]
> > === CONT  TestAcc_DataSourceMachine_Stable
> > cannot run Terraform provider tests: error calling terraform version command: fork/exec /tmp/plugintest-terraform272785777/terraform: text file busy
> > FAIL	github.com/juju/terraform-provider-juju/internal/provider	119.609s
> 
> That seems like there is a bug in terraform.

Yes so it seems. We'll have to solve it before making the terraform tests Required.

**Pull request merged**